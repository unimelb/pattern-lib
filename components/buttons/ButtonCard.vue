<template>
  <a
    class="btn-card"
    :href="link">
    <div
      class="btn-card__inner bg-white"
      :class="colorClass">
      <div
        v-if="topLabel"
        class="btn-card__top-label">{{ topLabel }}</div>
      <div class="btn-card__label">{{ label }}</div>
      <SvgIcon
        class="btn-card__icon"
        name="arrow-right"
        width="26px"
        height="22px"
        aria-hidden="true" />
    </div>
  </a>
</template>

<script>
import SvgIcon from '../icons/SvgIcon.vue';

export default {
  components: {
    SvgIcon,
  },
  props: {
    topLabel: {
      type: String,
      default: '',
    },
    label: {
      type: String,
      default: '',
      required: true,
    },
    link: {
      type: String,
      default: '',
      required: true,
    },
    inverted: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    colorClass() {
      return {
        'btn-card--inverted': this.inverted,
      };
    },
  },
};
</script>
