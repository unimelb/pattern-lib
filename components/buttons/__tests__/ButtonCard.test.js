import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import ButtonCard from '../ButtonCard.vue';

expect.extend(toHaveNoViolations);

describe('ButtonCard', () => {
  it('should match snapshot', () => {
    const result = shallow(ButtonCard, {
      propsData: {
        label: 'test label',
        link: 'www.unimelb.edu.au',
      },
    }).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(ButtonCard, {
      propsData: {
        label: 'test label',
        link: 'www.unimelb.edu.au',
      },
    }).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
