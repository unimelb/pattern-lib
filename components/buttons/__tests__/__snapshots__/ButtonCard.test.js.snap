// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonCard should match snapshot 1`] = `
<a
  class="btn-card"
  href="www.unimelb.edu.au"
>
  <div
    class="btn-card__inner bg-white"
  >
    <!---->
     
    <div
      class="btn-card__label"
    >
      test label
    </div>
     
    <!---->
  </div>
</a>
`;
